packages/eslint-plugin/src/utils/ast-utils.ts

```diff
@@ -81,6 +115,29 @@ function hasMemberExport(exportedMember, filePath) {
 }
 
 export function getRelativeImportPath(exportedMember, filePath, basePath) {
+  const TSX_SUFFIX = '.tsx';
+  const TS_SUFFIX = '.ts';
+  const JSX_SUFFIX = '.jsx';
+  const JS_SUFFIX = '.js';
+  const stats = lstatSync(filePath, { throwIfNoEntry: false }); // TODO nodejs bug: can return undefined despite contract https://nodejs.org/api/fs.html#fslstatsyncpath-options
+  if (!stats) {
+    const suffixesToTry = [TSX_SUFFIX, TS_SUFFIX, JSX_SUFFIX, JS_SUFFIX];
```

Comment:
Please replace this recursive logic with:
```ts
if (lstatSync(filePath).isDirectory()) {
  // existing unchanged code
} else if (!lstatSync(filePath, { throwIfNoEntry: false })){  // not folder, but not full file either
  // try to find an extension that exists
	const ext = ['.ts', '.tsx', '.js', '.jsx'].find(ext => lstatSync(filePath + ext, { throwIfNoEntry: false }));
  if (ext) {
    filePath += ext;
  }
}

// the rest of the logic unchanged
```

Reply:
Replaced